apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.ts

```diff
@@ -52,21 +53,31 @@ export class KnowledgeBoxProcessesComponent implements OnInit, OnDestroy {
 
   toggleLabelset(type: TrainingType, labelset: string) {
     if (this.selectedLabelsets[type].includes(labelset)) {
-      this.selectedLabelsets[type] = this.selectedLabelsets[type].filter((item) => item !== labelset)
-    }
-    else {
+      this.selectedLabelsets[type] = this.selectedLabelsets[type].filter((item) => item !== labelset);
+    } else {
       this.selectedLabelsets[type] = [...this.selectedLabelsets[type], labelset];
     }
     this.cdr?.markForCheck();
   }
 
+  toggleAll(type: TrainingType) {
+    this.labelsets.pipe(take(1)).subscribe((labelsets) => {
+      const selectAll = this.selectedLabelsets[type].length < labelsets.length;
+      this.selectedLabelsets[type] = selectAll ? labelsets.map((item) => item.value) : [];
+      this.cdr?.markForCheck();
```

Comment:
FYI pastanaga provides a helper function doing some extra checks. You can use it by doing `markForCheck(this.cdr)`.